kirby/api/__init__.py

```diff
@@ -15,36 +15,43 @@ class ClientError(BaseException):
 
 
 class Kirby:
-    def __init__(self, env_signature, session=None):
+    def __init__(self, env_signature, session=None, testing=False):
         ContextManager(env_signature)
         self.ctx = ctx
-        self._session = session or requests.session()
-
-        self._register()
+        self.testing = testing
+        if not testing:
+            self._session = session or requests.session()
+            self._register()
 
     def get_topic_id(self, topic_name):
-        result = self._session.get(
-            "/".join([self.ctx.KIRBY_WEB_SERVER, "topic"]),
-            params={"name": topic_name},
-        )
-        if result.status_code != 200:
-            if result.status_code == 500:
-                raise ServerError("There is an issue with the web server.")
-            else:
-                raise ClientError(
-                    f"There is no id from the name {topic_name}. "
-                    f"Verify that the topic has been registered."
-                )
-        return result.json()["id"]
+        if not self.testing:
+            result = self._session.get(
+                "/".join([self.ctx.KIRBY_WEB_SERVER, "topic"]),
+                params={"name": topic_name},
+            )
+            if result.status_code != 200:
+                if result.status_code == 500:
+                    raise ServerError("There is an issue with the web server.")
+                else:
+                    raise ClientError(
+                        f"There is no id from the name {topic_name}. "
+                        f"Verify that the topic has been registered."
```

Comment:
does the 2nd line **have** to be a f-string?

Reply:
Indeed it does not. Deleting f-string for the 2nd line